jx/ui/JXMLComponent.js

```diff
@@ -168,26 +181,28 @@ JXMLComponent.prototype.applyChildrenAttr = function(delta_children) {
 		dirty;
 
 	for (var k in delta_children) {
-		var child = delta_children[k];
+		var
+			child = delta_children[k],
+			child_uid = this.uid.replace(/:.*/, '') + ' ' + k;
 
 		if (!child) { // child is being removed
 			if (children[k]) {
 				// signal child for great destruction
-				this.children[k].destroy();
+				children[k].destroy();
```

Comment:
why do we call  `destroy()` on children here as well as in `onDIrty`? should it be done in one place?